collect_app/src/main/java/org/odk/collect/android/widgets/ExDecimalWidget.java

```diff
@@ -83,7 +83,7 @@ public ExDecimalWidget(Context context, FormEntryPrompt prompt) {
         	// truncate to 15 digits max...
             String dString = nf.format(d);
             d = Double.parseDouble(dString.replace(',', '.')); // in case , is decimal pt
-            mAnswer.setText(d.toString());
+            mAnswer.setText(String.format("%d", d.toString()));
```

Comment:
I think you need to remove the call on toString() for it to be an improvement. I'm surprised that this works -- %d doesn't require the param to be an int? I think %f is for a double.